app/models/concerns/latest_date_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Validates that a given date (attribute) does not occur after the date provided in options
+class LatestDateValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return unless options.key?(:date)
+
+    raise TypeError, "#{options[:date]} does not implement required comparison methods" unless options[:date].acts_like?(:date)
+
+    record.errors.add(attribute, "cannot occur before #{options[:date]}") unless valid?(value, options[:date])
```

Comment:
Should this be:
```suggestion
    record.errors.add(attribute, "cannot occur after #{options[:date]}") unless valid?(value, options[:date])
```

"after" instead of "before", since this checks if a value is after a latest possible date?

Reply:
In comparing this to #455, I've noticed that the error messages here follow a different style than the error messages added there. These error messages do not indicate the invalid attribute within the message, the error messages added in the API validation also contain the invalid value and attribute, for example: "'2020-13-13' is not a valid date for 'Date of Birth'". This was done to more closely match the style used in the `ImportController`. The API validation checks are only called from the API controller, so there shouldn't be an issue with these conflicting styles on import. 

But, the validators being added in this PR *will* get called from the API controller, and this could result in somewhat confusing error messages, since the API controller just takes the message as is. For example, a `date_of_birth` before `1900-01-01` on the API would just result in the message: `cannot occur before 1900-01-01`. No mention of the specific attribute that is an issue.

My current task is to refactor import validation, and I think as a part of that task I could update validation to use a consistent error style. To me it seems best to go with the style @Bialogs used here, not what is done on #455, since some of the conversation about logging original input values instead of normalized values and the differing needs of import validation and API validation make me think it is maybe best to separate attributes/values from the errors about them. It's too bad I didn't realize this earlier.

I don't want to hold this PR up though, so I'd propose we just live with the slightly unclear error messages for now until I add a PR for refactoring import validation, and clarify the error messages there. The API only takes in `date_of_birth`, `last_date_of_exposure` and `sympton_onset`, so it's not like it would be an insurmountable task to figure out which date is invalid if an API user got an error message like `cannot occur before 1900-01-01`. Let me know what you think @tstrass, @kylietmo, and @Bialogs. Sorry to have this realization minutes after #455 was merged to `0.16.0`, *not* the best timing.

Reply:
Another possible solution would be for me to make a PR for changing the error message style used in #455 to match what is used here. Would not be a difficult fix and could be done separately from the overall Import validation refactor.

Reply:
Nick and I just spoke about it and I agree that this is the preferrable way to do error messages.
>Another possible solution would be for me to make a PR for changing the error message style used in #455 to match what is used here. Would not be a difficult fix and could be done separately from the overall Import validation refactor.

We're moving forward with this plan